nullLane.go

```diff
@@ -176,3 +176,6 @@ func (nl *nullLane) SetPanicHandler(handler Panic) {
 func (wnw *wrappedNullWriter) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
+
+func (nl *nullLane) Metadata(key, value string){
```

Comment:
Hmm this is making me think.

IMO the Metadata should be a stand alone interface with only the metadata APIs, called `LaneMetadata`. And OpenSearchLane implements it. (NullLane or the other types do not.)

The caller should do this to set metadata:

```
// add metadata if the lane supports it
mdl := l.(lane.MetadataLane)
if mdl != nil {
    mdl.SetMetadata("email", token.email)
}
```

I think this is better than having to add Metadata to all the lanes, where all the others don't do anything with it.

Reply:
Sure, I can do this change